app/access.py

```diff
@@ -72,7 +72,14 @@ def wrapper(self, *args, **kwargs):
                 or "System admin" in self.current_user.permissions
             ):
                 raise tornado.web.HTTPError(401)
-            return method(self, *args, **kwargs)
+            print(method, args, kwargs)
+            try:
+                return method(self, *args, **kwargs)
+            except TypeError as e:
```

Comment:
Under what circumstance does the method return a TypeError?

I think this is in response to the request you had to 404 when `comment` instead of `comments` is specified as the URL, but I suspect it may be easier to specify that with a correct Tornado route?

Reply:
What does that mean? Right now the app fails with a TypeError if someone messes up.

Reply:
If the route specification in the app factory is specific enough, why does it get sent to the route in the first place?

Reply:
I.e., where does the TypeError come from?

Reply:
I don't know what you mean. I just see that error when calling the method. The issue I referenced showed the error message.